src/processTargets/modifiers/getContainingScope.ts

```diff
@@ -0,0 +1,30 @@
+import { Direction } from "../../typings/targetDescriptor.types";
+import { TextEditor, Position } from "vscode";
+import { ScopeHandler } from "./scopeHandlers/scopeHandler.types";
+
+/**
+ * Gets the smallest containing scope, preferring scopes in direction
+ * {@link direction}
+ * @param scopeHandler
+ * @param editor
+ * @param position
+ * @param direction
+ * @returns
+ */
+export function getContainingScope(
+  scopeHandler: ScopeHandler,
+  editor: TextEditor,
+  position: Position,
+  direction: Direction,
+) {
+  return getOne(
+    scopeHandler.generateScopes(editor, position, direction, {
+      containment: "required",
+    }),
+  );
+}
+
+function getOne<T>(iterable: Iterable<T>): T | undefined {
+  const { value, done } = iterable[Symbol.iterator]().next();
```

Comment:
Depending on if the generator is using a return statement or not the very last value could have `done: true`
Calling next after the generator is done will just make the value undefined.
In this case I would just return the value and don't care about `done`